### DISCUSSION TITLE
Getting "dependency has no live implementation" warning for DependencyKey-conforming object

### DISCUSSION THREAD

**Discussion by acosmicflamingo** [pointfreeco/swift-composable-architecture#1745] [Q&A] [answered]
I'm quite puzzled by what the runtime warning `"@Dependency(\.localPersistenceClient)" has no live implementation, but was accessed from a live context` means when my dependency does actually conform to DependencyKey:

```swift
extension LocalPersistenceClient: DependencyKey {
  private static let userSettingsKey = "co.pointfree.isowords.PersistenceClient.userSettings"

  public static var liveValue: LocalPersistenceClient {
    let defaultUserSettings = UserSettings()
    var userSettings = UserDefaults.standard.data(forKey: userSettingsKey)
      .flatMap({ try? decoder.decode(UserSettings.self, from: $0) }) ?? defaultUserSettings
    {
      didSet {
        UserDefaults.standard.set(
          try? encoder.encode(userSettings),
          forKey: userSettingsKey
        )
      }
    }

    return Self(
      userSettings: { userSettings },
      setUserSettings: { newUserSettings in
        userSettings = newUserSettings
      }
    )
  }
}
```

This error goes away if I explicitly set it in my AppReducer, but why is this even needed?
```swift
@main class AppDelegate: UIResponder, UIApplicationDelegate {
  let store = Store(
    initialState: .init(),
    reducer: AppMainReducer().transformDependency(\.self) {
      $0.localPersistenceClient = .liveValue
    }
  )
}
```

Isowords does something similar but only for 3 dependencies: https://github.com/pointfreeco/isowords/blob/main/App/iOS/App.swift#L15. What makes those 3 dependencies special enough to require being explicitly set and not the many, many other dependencies that exist in the app?

**Comment by tgrapperon** [comment, marked as answer]
Hey @acosmicflamingo! This is maybe related to #1640. Maybe you can try to reference `.liveValue` in the same file as `AppDelegate` (even without assigning it to your reducer). This is not optimal, but it's maybe enough for the compiler to pick it up.

**Comment by acosmicflamingo** [comment]
Hi @tgrapperon ! Oh that does seems exactly like what is happening with me :) I'm using Tuist to handle my app/module/etc, so this makes sense. Thank you!
